examples/kalman_filter.py

```diff
@@ -0,0 +1,416 @@
+
+The former is a 1D-localization problem where the state is a 2D vector (x, v)
+made of the system's position and speed. The process writes
+:math:`(x_{i+1}, v_{i+1}) = (x_i + dt * v_i, v_i + dt * a_i)`,
+where dt is the time-step between i and i+1, and a_i a noisy acceleration
+measured by a given sensor.
+
+The latter is a 2D pose (position + orientation) estimation problem, where the
+state (R, x) is made a planar rotation and 2D position, i.e. a member of SE(2).
+The non-linear propagation writes
+:math:`(R_{i+1}, x_{i+1}) = (R_i \Omega_i, x_i + dt * R_i u_i)`,
+where :math:`\Omega_i, u_i` is the measured odometry of the system.
+The implementation follows that of the Invariant Extended Kalman Filter (IEKF)
+which was designed for such cases [BB2017].
+
+References
+----------
+.. [BB2017] Barrau, Bonnabel, "The Invariant Extended Kalman Filter as a Stable
+Observer", IEEE Transactions on Automatic Control, 2017
+https://arxiv.org/abs/1410.1465
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import geomstats.backend as gs
+from geomstats import algebra_utils
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.matrices import Matrices
+from geomstats.geometry.special_euclidean import SpecialEuclidean
+
+
+class LocalizationLinear:
+    """Class for modeling a linear 1D localization problem.
+
+    The state is made of a scalar position and scalar speed, thus a 2D vector.
+    A sensor provides acceleration inputs, while another one provides sparse
+    measurements of the position.
+    """
+
+    group = Euclidean(2)
+    dim = group.dim
+    dim_noise = 1
+    dim_obs = 1
+
+    @staticmethod
+    def propagate(state, sensor_input):
+        """Propagate with piece-wise constant acceleration and velocity."""
+        dt, acc = sensor_input
+        pos, speed = state
+        pos = pos + dt * speed
+        speed = speed + dt * acc
+        return gs.array([pos, speed])
+
+    @staticmethod
+    def propagation_jacobian(state, sensor_input):
+        """Compute the matrix associated to the affine propagation."""
+        dt, _ = sensor_input
+        dim = LocalizationLinear.dim
+        position_line = gs.hstack((gs.eye(dim // 2), dt * gs.eye(dim // 2)))
+        speed_line = gs.hstack((
+            gs.zeros((dim // 2, dim // 2)), gs.eye(dim // 2)))
+        jac = gs.vstack((position_line, speed_line))
+        return jac
+
+    @staticmethod
+    def noise_jacobian(state, sensor_input):
+        """Compute the matrix associated to the propagation noise.
+
+        The noise is supposed additive.
+        """
+        dt, _ = sensor_input
+        dim = LocalizationLinear.dim
+        position_wrt_noise = gs.zeros((dim // 2, dim // 2))
+        speed_wrt_noise = gs.sqrt(dt) * gs.eye(dim // 2)
+        jac = gs.vstack((position_wrt_noise, speed_wrt_noise))
+        return jac
+
+    @staticmethod
+    def observation_jacobian(state, observation):
```

Comment:
`observation` unused?

Reply:
In this case yes, but since in other cases it might, I preferred keeping it here so that if one changes the model the KalmanFilter class would still work. Don't really know whether it's best practice, but I could not think of an alternative yet.